src/System.Slices/System/Buffers/OwnedMemory.cs

```diff
@@ -120,8 +122,8 @@ protected internal virtual DisposableReservation Reserve(ref ReadOnlyMemory<T> m
         #region Used by Memory<T>
         void IKnown.AddReference(long id)
         {
-            VerifyId(id);
             AddReference();
+            VerifyId(id);
```

Comment:
Let's rethrow, i.e. I think it's ok that we erase the callstack in this case as there is no user code on it ever.